File: Answer1-3-4/Refactor-Code.go
package main

import (
	"fmt"
	"strings"
)

func findFirstStringInBracket(str string) string {
	if len(str) > 0 {
		indexFirstBracket := strings.Index(str, "(")
		if indexFirstBracket >= 0 {
			wordsAfterFirstBracket := str[indexFirstBracket+1:]
			indexClosingBracketFound := strings.Index(wordsAfterFirstBracket, ")")
			if indexClosingBracketFound >= 0 {
				return wordsAfterFirstBracket[:indexClosingBracketFound]
			}
			return ""
		}
	}
	return ""
}

func main() {
	s := "Muhammad Nur )))(Fadli)"
	fmt.Println(findFirstStringInBracket(s))
}

File: Answer1-3-4/Refactor-Code.php
<?php
function findFirstStringInBracket($str) {

    if(strlen($str) > 0) {
        $firstbracket = strstr($str, '(');
        if($firstbracket) {
            $firstbracket = ltrim($firstbracket, '(');
            return strstr($firstbracket, ')', true);
        }
        return '';
    }
    return '';
}

echo findFirstStringInBracket("Muhammad Nur )))(Fadli)");
?>